asserts/extkeypairmgr.go

```diff
@@ -0,0 +1,260 @@
+	err := em.keyMgr("get-public-key", []string{"-f", "DER", "-k", name}, nil, &k)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot find external key: %v", err)
+	}
+	pubk, err := x509.ParsePKIXPublicKey(k)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot decode external key %q: %v", name, err)
+	}
+	rsaPub, ok := pubk.(*rsa.PublicKey)
+	if !ok {
+		return nil, nil, fmt.Errorf("expected RSA public key, got instead: %T", pubk)
+	}
+	pubKey := RSAPublicKey(rsaPub)
+	return pubKey, rsaPub, nil
+}
+
+func (em *ExternalKeypairManager) Export(keyName string) ([]byte, error) {
+	pubKey, _, err := em.findByName(keyName)
+	if err != nil {
+		return nil, err
+	}
+	return EncodePublicKey(pubKey)
+}
+
+func (em *ExternalKeypairManager) loadKey(name string) (*cachedExtKey, error) {
+	id, ok := em.nameToID[name]
+	if ok {
+		return em.cache[id], nil
+	}
+	pubKey, rsaPub, err := em.findByName(name)
+	if err != nil {
+		return nil, err
+	}
+	id = pubKey.ID()
+	em.nameToID[name] = id
+	cachedKey := &cachedExtKey{
+		pubKey: pubKey,
+		signer: &extSigner{
+			keyName: name,
+			rsaPub:  rsaPub,
+			// signWith is filled later
+		},
+	}
+	em.cache[id] = cachedKey
+	return cachedKey, nil
+}
+
+func (em *ExternalKeypairManager) privateKey(cachedKey *cachedExtKey) PrivateKey {
+	if cachedKey.privKey == nil {
+		extSigner := cachedKey.signer
+		// fill signWith
+		extSigner.signWith = em.signWith
+		signer := packet.NewSignerPrivateKey(v1FixedTimestamp, extSigner)
+		signk := openpgpPrivateKey{privk: signer}
+		extKey := &extPGPPrivateKey{
+			pubKey:     cachedKey.pubKey,
+			from:       fmt.Sprintf("external keypair manager %q", em.keyMgrPath),
+			externalID: extSigner.keyName,
+			bitLen:     extSigner.rsaPub.N.BitLen(),
+			doSign:     signk.sign,
+		}
+		cachedKey.privKey = extKey
+	}
+	return cachedKey.privKey
+}
+
+func (em *ExternalKeypairManager) GetByName(keyName string) (PrivateKey, error) {
+	cachedKey, err := em.loadKey(keyName)
+	if err != nil {
+		return nil, err
+	}
+	return em.privateKey(cachedKey), nil
+}
+
+func (em *ExternalKeypairManager) Put(privKey PrivateKey) error {
+	return fmt.Errorf("cannot import private key into external keypair manager")
+}
+
+func (em *ExternalKeypairManager) loadAllKeys() error {
+	names, err := em.keyNames()
+	if err != nil {
+		return err
+	}
+	for _, name := range names {
+		if _, err := em.loadKey(name); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (em *ExternalKeypairManager) Get(keyID string) (PrivateKey, error) {
+	cachedKey, ok := em.cache[keyID]
+	if !ok {
+		// try to load all keys
+		if err := em.loadAllKeys(); err != nil {
+			return nil, err
+		}
+		cachedKey, ok = em.cache[keyID]
+		if !ok {
+			return nil, fmt.Errorf("cannot find external key with id %q", keyID)
+		}
+	}
+	return em.privateKey(cachedKey), nil
+}
+
+var digestInfoSHA512Prefix = []byte{0x30, 0x51, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x03, 0x05, 0x00, 0x04, 0x40}
```

Comment:
(nitpick) maybe a link to the relevant spec or resource where this header is speced? i.e. to give a hint where this comes from?

Reply:
found something though it's a bit of a maze